fe/fe-core/src/main/java/com/starrocks/load/InsertOverwriteJobRunner.java

```diff
@@ -556,10 +566,51 @@ private void doCommit(boolean isReplay) {
                 throw new DdlException("partition type " + partitionInfo.getType() + " is not supported");
             }
 
-            long sumTargetRows = job.getTmpPartitionIds().stream()
-                    .mapToLong(p -> targetTable.mayGetPartition(p).stream().mapToLong(Partition::getRowCount).sum())
-                    .sum();
+            long sumTargetRows = 0;
+            if (insertStmt != null && !isReplay) {
+                TransactionState txnState = GlobalStateMgr.getCurrentState()
+                        .getGlobalTransactionMgr()
+                        .getTransactionState(dbId, insertStmt.getTxnId());
+                if (txnState != null) {
+                    // Get target rows from TxnCommitAttachment
+                    if (txnState.getTxnCommitAttachment() != null) {
+                        TxnCommitAttachment attachment = txnState.getTxnCommitAttachment();
+                        if (attachment instanceof InsertTxnCommitAttachment) {
+                            sumTargetRows = ((InsertTxnCommitAttachment) attachment).getLoadedRows();
+                        }
+                    }
+                    
+                    // Collect tablet row counts from TableCommitInfo for statistics sampling
+                    TableCommitInfo tableCommitInfo = txnState.getIdToTableCommitInfos().get(tableId);
+                    if (tableCommitInfo != null) {
+                        for (var entry : tableCommitInfo.getIdToPartitionCommitInfo().entrySet()) {
+                            long physicalPartitionId = entry.getKey();
+                            PartitionCommitInfo partitionCommitInfo = entry.getValue();
+                            Map<Long, Long> tabletRows = partitionCommitInfo.getTabletIdToRowCountForPartitionFirstLoad();
+                            
+                            PhysicalPartition physicalPartition = targetTable.getPhysicalPartition(physicalPartitionId);
+                            if (physicalPartition != null) {
+                                Partition partition = targetTable.getPartition(physicalPartition.getParentId());
+                                if (partition != null && !tabletRows.isEmpty()) {
+                                    long partitionId = partition.getId();
+                                    tabletRows.forEach((tabletId, rowCount) -> partitionTabletRowCounts.put(partitionId,
+                                            tabletId, rowCount));
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            
+            if (sumTargetRows == 0) {
+                LOG.warn("TxnCommitAttachment is null or invalid, fallback to partition.getRowCount()");
```

Comment:
it's better to attach the context like table_id/partition_id/tablet_id/txn_id ?

Reply:
done